pkg/api/types.go

```diff
@@ -320,8 +320,48 @@ type Result struct {
 
 	// ImageID describes resulting image ID.
 	ImageID string
+
+	// BuildInfo holds information about the result of a build.
+	BuildInfo []*BuildStepInfo
+}
+
+type BuildStepInfo struct {
+	// FailureReason is a camel case reason that is used by the machine to reply
+	// back to the OpenShift builder with information as why any of the steps in
+	// the build, failed.
+	FailureReason StepFailureReason
+
+	// FuncDuration holds the elapsed time for a specific build step operation
+	// (git  clone, assemble container, etc.) in nanoseconds.
+	FuncDuration int64
```

Comment:
i guess openshift does not call each s2i step (clone, assemble, commit) separately, but it does call clone explicitly.  If we need s2i to record and report the assemble+commit time then i guess we'll have to do that, but again let's do it in a separate PR after this card is complete.